src/main/java/org/roaringbitmap/RunContainer.java

```diff
@@ -496,6 +522,38 @@ public Container and(BitmapContainer x) {
         else return answer.toArrayContainer();
     }
 
+    @Override
+    public int andCardinality(BitmapContainer x) {
+        // could be implemented as return toBitmapOrArrayContainer().iand(x);
+        int card = this.getCardinality();
+        int cardinality=0;
+        if (card <=  ArrayContainer.DEFAULT_MAX_SIZE) {
+            // result can only be an array (assuming that we never make a RunContainer)
+            if(card > x.cardinality) card = x.cardinality;
+            cardinality=0;
+            for (int rlepos=0; rlepos < this.nbrruns; ++rlepos) {
+                int runStart = Util.toIntUnsigned(this.getValue(rlepos));
+                int runEnd = runStart + Util.toIntUnsigned(this.getLength(rlepos));
+                for (int runValue = runStart; runValue <= runEnd; ++runValue) {
+                    if ( x.contains((short) runValue)) {// it looks like contains() should be cheap enough if accessed sequentially
+                        cardinality++;
+                    }
+                }
+            }
+            return cardinality;
+        }
+        // we expect the answer to be a bitmap (if we are lucky)
+        BitmapContainer answer = x.clone();
```

Comment:
I believe it will be simpler and more efficient (because the bitmap container `x` is cloned here) to always use the algorithm in the if block above.